docs/DeveloperGuide.md

```diff
@@ -324,14 +324,34 @@ The following sequence diagram shows how the `RemoveCommand` is executed:
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+This product helps users streamline the process of tracking and managing their internship applications by allowing them to:
+
+- **Store Important Internship Information**: Users can input and store key details such as the role, company, duration, required skills, application deadlines, and any additional notes regarding each internship.
+
+- **Track Application Status**: Users can keep track of the current status of their applications (e.g., not applied, applied, interview scheduled, offer received) and update it as the application progresses.
+
+- **Edit and Manage Internships**: Users can edit saved internships, modify application statuses, update deadlines, or remove outdated internships from their list.
+
+- **Filter Internships by Key Criteria**: Users can filter tracked internships by relevant criteria such as required skills, application status, or upcoming deadlines, helping them find specific applications.
+
+- **Sort Internships by Key Criteria**: Users can sort tracked internships by relevant criteria such as required skills, application status, or upcoming deadlines, helping them prioritize applications.
+
+- **Monitor Deadlines Easily**: Users can stay on top of their internship deadlines by sorting or flagging applications with approaching deadlines, ensuring that no important opportunity is missed.
```

Comment:
I like this draft! We will definitely have to make some changes but its a good start.